src/instrumentations/index.ts

```diff
@@ -233,3 +260,82 @@ export function getInstrumentations() {
 
   return loaded;
 }
+
+export function configureInstrumentations(options: Options) {
+  const instrumentations = options.tracing.instrumentations || [];
+  for (const instrumentation of instrumentations) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const instr = instrumentation as any;
+
+    switch (instr['instrumentationName']) {
+      case '@opentelemetry/instrumentation-graphql':
+        configureGraphQlInstrumentation(instr, options.tracing);
+        break;
+      case '@opentelemetry/instrumentation-http':
+        configureHttpInstrumentation(instr, options.tracing);
+        break;
+      case '@opentelemetry/instrumentation-redis':
+        configureRedisInstrumentation(instr, options.tracing);
+        break;
+      case '@opentelemetry/instrumentation-bunyan':
+        disableLogSending(instr);
+        configureLogInjection(instr);
+        break;
+      case '@opentelemetry/instrumentation-pino':
+      case '@opentelemetry/instrumentation-winston':
+        configureLogInjection(instr);
+        break;
+    }
+  }
+}
+
+export function parseOptionsAndConfigureInstrumentations(
+  options: Partial<StartOptions> = {}
+) {
+  const { metrics, profiling, tracing, logging, ...restOptions } = options;
+
+  assertNoExtraneousProperties(restOptions, [
+    'accessToken',
+    'endpoint',
+    'serviceName',
+    'logLevel',
+  ]);
+
+  const startProfilingOptions = Object.assign(
+    pick(restOptions, allowedProfilingOptions),
+    profiling
+  );
+
+  assertNoExtraneousProperties(startProfilingOptions, allowedProfilingOptions);
+  const profilingOptions = setDefaultProfilingOptions(startProfilingOptions);
+
+  const startLoggingOptions = Object.assign(
+    pick(restOptions, allowedLoggingOptions),
+    logging
+  );
+
+  const loggingOptions = setDefaultLoggingOptions(startLoggingOptions);
+
+  const startTracingOptions = Object.assign(
+    pick(restOptions, allowedTracingOptions),
+    tracing
+  ) as Partial<TracingOptions>;
+
+  assertNoExtraneousProperties(startTracingOptions, allowedTracingOptions);
+  const tracingOptions = setDefaultTracingOptions(startTracingOptions);
+
+  const startMetricsOptions = Object.assign(
+    pick(restOptions, allowedMetricsOptions),
+    metrics
+  );
+
+  assertNoExtraneousProperties(startMetricsOptions, allowedMetricsOptions);
+  const metricsOptions = setDefaultMetricsOptions(startMetricsOptions);
+
+  configureInstrumentations({
```

Comment:
Would it make sense to split `configureInstrumentations` from this function and rename it just `parseOptions`?